salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+
+        Arguments:
+            client {hvac} -- hvac client
+            ret {dict} -- salt state result
+
+        Returns:
+            list -- secrets engines
+        """
+        log.info('Retrieving secrets engines from Vault')
+        remote_secret_engines = []
+        try:
+            secrets_engines_resp = client.sys.list_mounted_secrets_engines()
+            for engine in secrets_engines_resp['data']:
+                remote_secret_engines.append(
+                    VaultSecretEngine(
+                        type=secrets_engines_resp[engine]['type'],
+                        path=(secrets_engines_resp[engine]["path"]
+                              if 'path' in secrets_engines_resp[engine] else engine),
+                        description=secrets_engines_resp[engine]["description"],
+                        config=OrderedDict(
+                            sorted(secrets_engines_resp[engine]["config"].items()))
+                    )
+                )
+            remote_secret_engines.sort(key=lambda x: x.type)
+        except Exception:
+            raise
+
+        log.info('Finished retrieving secrets engines from vault.')
+        return remote_secret_engines
+
+    def populate_local_secrets_engines(self, configs, ret):
+        """Retriev secrets engines from local config
+
+        Arguments:
+            configs {list} -- local secrets engines information
+            ret {dict} -- salt state result
+
+        Returns:
+            list -- secrets engines
+        """
+        log.info('Populating local secret engines...')
+        local_secret_engines = []
+        try:
+            for secret_engine in configs:
+                config = None
+                secret_config = None
+                extra_config = None
+
+                if 'config' in secret_engine:
+                    if secret_engine["config"]:
+                        config = OrderedDict(
+                            sorted(secret_engine["config"].items()))
+
+                if 'secret_config' in secret_engine:
+                    if secret_engine["secret_config"]:
+                        secret_config = OrderedDict(
+                            sorted(secret_engine["secret_config"].items()))
+
+                if 'extra_config' in secret_engine:
+                    if secret_engine["extra_config"]:
+                        extra_config = OrderedDict(
+                            sorted(secret_engine["extra_config"].items()))
+
+                local_secret_engines.append(VaultSecretEngine(
+                    type=secret_engine["type"],
+                    path=secret_engine["path"],
+                    description=secret_engine["description"],
+                    config=config,
+                    secret_config=secret_config,
+                    extra_config=extra_config
+                ))
+
+            local_secret_engines.sort(key=lambda x: x.type)
+        except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.